pypeit/find_objects.py

```diff
@@ -960,21 +957,275 @@ class SlicerIFUFindObjects(MultiSlitFindObjects):
 
-    def find_objects_pypeline(self, image, ivar, std_trace=None,
-                              show_peaks=False, show_fits=False, show_trace=False,
-                              show=False, save_objfindQA=False, neg=False, debug=False,
-                              manual_extract_dict=None):
+    def initialize_slits(self, slits, initial=True):
+        """
+        Gather all the :class:`~pypeit.slittrace.SlitTraceSet` attributes
+        that we'll use here in :class:`FindObjects. Identical to the parent but the slits are not trimmed.
+
+
+        Args:
+            slits (:class:`~pypeit.slittrace.SlitTraceSet`):
+                SlitTraceSet object containing the slit boundaries that will be initialized.
+            initial (:obj:`bool`, optional):
+                Use the initial definition of the slits. If False,
+                tweaked slits are used.
+        """
+        super().initialize_slits(slits, initial=True)
+
+    def global_skysub(self, skymask=None, update_crmask=True,
+                      previous_sky=None, show_fit=False, show=False, show_objs=False, objs_not_masked=False,
+                      reinit_bpm: bool = True):
+        """
+        Perform global sky subtraction. This SlicerIFU-specific routine ensures that the
+        edges of the slits are not trimmed, and performs a spatial and spectral
+        correction using the sky spectrum, if requested. See Reduce.global_skysub()
+        for parameter definitions.
+
+        See base class method for description of parameters.
+
+        Args:
+            reinit_bpm (:obj:`bool`, optional):
+                If True (default), the bpm is reinitialized to the initial bpm
+                Should be False on the final run in case there was a failure
+                upstream and no sources were found in the slit/order
+        """
+
+        global_sky_sep = super().global_skysub(skymask=skymask, update_crmask=update_crmask,
+                                               previous_sky=previous_sky, show_fit=show_fit, show=show,
+                                               show_objs=show_objs,
+                                               objs_not_masked=objs_not_masked, reinit_bpm=reinit_bpm)
+
+        # Check if flexure or a joint fit is requested. If not return this
+        if not self.par['reduce']['skysub']['joint_fit'] and self.par['flexure']['spec_method'] == 'skip':
+            return global_sky_sep
+
+        if self.wv_calib is None:
+            msgs.error("A wavelength calibration is needed (wv_calib) if a joint sky fit is requested.")
+        msgs.info("Generating wavelength image")
+
+        # Generate the waveimg which is needed if flexure is being computed
+        self.waveimg = self.wv_calib.build_waveimg(self.tilts, self.slits, spat_flexure=self.spat_flexure_shift)
+        # TODO: Can the flexure compensation be removed from this class or is it necessary for the joint skysub?
```

Comment:
It is necessary for the joint skysub (you need the wavelength to be correct to fit all slices simultaneously). The idea is that a model of the sky spectrum is computed, and this is used to perform a relative flexure of all slices. Then, the "absolute" flexure is computed with a single (reference) slice and applied to all slices.

I think it can be (eventually) moved to the flexure code, but that needs a lot of tidying up first.